src/jdk.incubator.vector/share/classes/jdk/incubator/vector/Float16.java

```diff
@@ -0,0 +1,1518 @@
+     *      to be greater than negative zero.
+     * </ul>
+     *
+     * @param   anotherFloat16   the {@code Float16} to be compared.
+     * @return  the value {@code 0} if {@code anotherFloat16} is
+     *          numerically equal to this {@code Float16}; a value
+     *          less than {@code 0} if this {@code Float16}
+     *          is numerically less than {@code anotherFloat16};
+     *          and a value greater than {@code 0} if this
+     *          {@code Float16} is numerically greater than
+     *          {@code anotherFloat16}.
+     *
+     * @see Float#compareTo(Float)
+     * @see Double#compareTo(Double)
+     * @jls 15.20.1 Numerical Comparison Operators {@code <}, {@code <=}, {@code >}, and {@code >=}
+     */
+    @Override
+    public int compareTo(Float16 anotherFloat16) {
+        return compare(this, anotherFloat16);
+    }
+
+    /**
+     * Compares the two specified {@code Float16} values.
+     *
+     * @param   f1        the first {@code Float16} to compare
+     * @param   f2        the second {@code Float16} to compare
+     * @return  the value {@code 0} if {@code f1} is
+     *          numerically equal to {@code f2}; a value less than
+     *          {@code 0} if {@code f1} is numerically less than
+     *          {@code f2}; and a value greater than {@code 0}
+     *          if {@code f1} is numerically greater than
+     *          {@code f2}.
+     *
+     * @see Float#compare(float, float)
+     * @see Double#compare(double, double)
+     */
+    public static int compare(Float16 f1, Float16 f2) {
+        return Float.compare(f1.floatValue(), f2.floatValue());
+    }
+
+    /**
+     * Returns the larger of two {@code Float16} values.
+     *
+     * @apiNote
+     * This method corresponds to the maximum operation defined in
+     * IEEE 754.
+     *
+     * @param a the first operand
+     * @param b the second operand
+     * @return the greater of {@code a} and {@code b}
+     * @see java.util.function.BinaryOperator
+     * @see Math#max(float, float)
+     * @see Math#max(double, double)
+     */
+    public static Float16 max(Float16 a, Float16 b) {
+        return shortBitsToFloat16(floatToFloat16(Math.max(a.floatValue(),
+                                                          b.floatValue() )));
+    }
+
+    /**
+     * Returns the smaller of two {@code Float16} values.
+     *
+     * @apiNote
+     * This method corresponds to the minimum operation defined in
+     * IEEE 754.
+     *
+     * @param a the first operand
+     * @param b the second operand
+     * @return the smaller of {@code a} and {@code b}
+     * @see java.util.function.BinaryOperator
+     * @see Math#min(float, float)
+     * @see Math#min(double, double)
+     */
+    public static Float16 min(Float16 a, Float16 b) {
+        return shortBitsToFloat16(floatToFloat16(Math.min(a.floatValue(),
+                                                          b.floatValue()) ));
```

Comment:
I assume we will optimize these min/max implementations in the future. Otherwise, the extra space should be removed.

Reply:
The general intention is that the back-end intrinsification efforts that use native hardware instructions, where present, will provide the main speed boost. Therefore, the preference has been to keep the Java sources in Float16 as straightforward and "obviously correct" as possible.